tests/test_bootstrap.py

```diff
@@ -0,0 +1,1390 @@
+        """Try optimizing for bootstrap consistency in axisymmetry, at fixed shape.
+
+        This version of the test covers an equilibrium with an iota
+        profile rather than a current profile.
+        """
+        ne0 = 4.0e20
+        T0 = 12.0e3
+        ne = PowerSeriesProfile(ne0 * np.array([1, -1]), modes=[0, 10])
+        Te = PowerSeriesProfile(T0 * np.array([1, -1]), modes=[0, 2])
+        Ti = Te
+        Zeff = 1
+
+        helicity = (1, 0)
+        B0 = 5.0  # Desired average |B|
+        LM_resolution = 8
+
+        initial_iota = -0.61
+        num_iota_points = 21
+        iota = SplineProfile(np.full(num_iota_points, initial_iota))
+
+        # Set initial condition:
+        Rmajor = 6.0
+        aminor = 2.0
+        NFP = 1
+        surface = FourierRZToroidalSurface(
+            R_lmn=np.array([Rmajor, aminor]),
+            modes_R=[[0, 0], [1, 0]],
+            Z_lmn=np.array([-aminor]),
+            modes_Z=[[-1, 0]],
+            NFP=NFP,
+        )
+
+        eq = Equilibrium(
+            surface=surface,
+            electron_density=ne,
+            electron_temperature=Te,
+            ion_temperature=Ti,
+            atomic_number=Zeff,
+            iota=iota,
+            Psi=B0 * np.pi * (aminor**2),
+            NFP=NFP,
+            L=LM_resolution,
+            M=LM_resolution,
+            N=0,
+            L_grid=2 * LM_resolution,
+            M_grid=2 * LM_resolution,
+            N_grid=0,
+            sym=True,
+        )
+
+        eq.solve(
+            verbose=3,
+            ftol=1e-8,
+            constraints=get_fixed_boundary_constraints(kinetic=True),
+            optimizer=Optimizer("lsq-exact"),
+            objective=ObjectiveFunction(objectives=ForceBalance()),
+        )
+
+        eq.save("test_bootstrap_consistency_iota_initial.h5")
+        # eq = desc.io.load("test_bootstrap_consistency_iota_initial.h5")
+
+        # Done establishing the initial condition. Now set up the optimization.
+
+        constraints = (
+            ForceBalance(),
+            FixBoundaryR(),
+            FixBoundaryZ(),
+            FixElectronDensity(),
+            FixElectronTemperature(),
+            FixIonTemperature(),
+            FixAtomicNumber(),
+            FixPsi(),
+        )
+
+        # grid for bootstrap consistency objective:
+        grid = LinearGrid(
+            rho=np.linspace(1e-4, 1 - 1e-4, (num_iota_points - 1) * 2 + 1),
+            M=eq.M * 2,
+            N=eq.N * 2,
+            NFP=eq.NFP,
+        )
+        objective = ObjectiveFunction(
+            BootstrapRedlConsistency(
+                grid=grid,
+                helicity=helicity,
+            )
+        )
+        eq, _ = eq.optimize(
+            verbose=3,
+            objective=objective,
+            constraints=constraints,
+            optimizer=Optimizer("scipy-trf"),
+            ftol=1e-6,
+        )
+
+        eq.save("test_bootstrap_consistency_iota_final.h5")
```

Comment:
same note about saving to temp directory

Reply:
Now `TmpDir` is used.